### PR TITLE
Update CRDB versions

### PR DESCRIPTION
**Pull request by pseudomuto**
Closes #694 

I've updated the CRDB versions here and excluded v21.1.8 which prevented downgrades.

### CODE REVIEW

**Review by davidwding — APPROVED** [review]

File: examples/example.yaml [hunk 1]
```diff
@@ -40,9 +40,9 @@ spec:
       memory: 8Gi
   tlsEnabled: true
 # You can set either a version of the db or a specific image name
-# cockroachDBVersion: v21.1.7
+# cockroachDBVersion: v21.1.9
   image:
-    name: cockroachdb/cockroach:v21.1.7
+    name: cockroachdb/cockroach:v21.1.9
```
Comment (chrislovecnm) [inline examples/example.yaml:45]: Did you modify the template that generates this file? Most of our YAML files are generated to update with the latest version.

File: hack/update-crdb-versions.sh [hunk 2]
```diff
@@ -47,8 +51,14 @@ cat > crdb-versions.yaml << EOF
 
 CrdbVersions:
 EOF
+}
+
+dump_versions() {
+  local nameFilter=".data[] .repositories[] .tags[] .name"
+  # Skip unsupported versions and the latest tag
+  for version in $(curl -s $URL | jq -r "${nameFilter}" | grep -v ^v19 | grep -v ^v21.1.8$ | grep -v latest | grep -v ubi$ | sort --version-sort); do
```
Comment (chrislovecnm) [inline hack/update-crdb-versions.sh:59]: This is a pretty complicated command that can fail a lot of ways.  Including jq is not installed.  Should we break this into more steps and check that the data is returned is valid?

**Review by chrislovecnm — CHANGES_REQUESTED** [review]
See comments

File: hack/update-crdb-versions.sh (hunk 2 above)
Comment (pseudomuto) [reply hack/update-crdb-versions.sh:59]: I agree about breaking this up. However, this is in place now. Meaning jq is installed in the CI environment where the PR bot runs.

I was talking with @davidwding earlier about converting some of these critical bash scripts into some nice, testable, go apps. A problem for another PR though.

Comment (udnay) [reply hack/update-crdb-versions.sh:59]: I agree, I think we can file an issue and ship this

Comment (pseudomuto) [reply hack/update-crdb-versions.sh:59]: Issue logged https://github.com/cockroachdb/cockroach-operator/issues/738

**Review by davidwding — APPROVED** [review]

File: hack/update-crdb-versions.sh (hunk 2 above)
Comment (pseudomuto) [reply hack/update-crdb-versions.sh:59]: @chrislovecnm you good with this now?

**Review by chrislovecnm — APPROVED** [review]

**Comment by chrislovecnm** [comment]
LGTM

**Review by davidwding — APPROVED** [review]

**Pull request merged**